src/scripts/sync_subclassof.py

```diff
@@ -0,0 +1,117 @@
+"""Create a robot template for purpose of syncing subclassof relations.
+
+Resources
+- GitHub issue: https://github.com/monarch-initiative/mondo-ingest/issues/92
+- GitHub PR: https://github.com/monarch-initiative/mondo-ingest/pull/363
+"""
+import os
+from argparse import ArgumentParser
+from pathlib import Path
+from typing import Any, Dict, List, Tuple, Union
+
+import pandas as pd
+from oaklib import get_adapter
+from oaklib.interfaces.basic_ontology_interface import BasicOntologyInterface, RELATIONSHIP
+from oaklib.types import CURIE, URI
+
+from src.scripts.utils import DEFAULT_PREFIXES_CSV, get_monarch_curies_converter, remove_angle_brackets
+
+HERE = Path(os.path.abspath(os.path.dirname(__file__)))
+SRC_DIR = HERE.parent
+ONTOLOGY_DIR = SRC_DIR / 'ontology'
+DEFAULT_OUTPATH = ONTOLOGY_DIR / 'reports' / 'sync-subclassof.robot.template.tsv'
+DEFAULT_ONTOLOGY_DB_PATH = ONTOLOGY_DIR / 'tmp' / 'merged.db'
+ROBOT_ROW = [{
+    'term_id': 'ID',
+    'parent_id': 'SC % SPLIT=|',
+    'xref_ids': 'A oboInOwl:hasDbXref SPLIT=|',  # todo: Nico may provide different value
```

Comment:
Ahh, I see. Will do this. This clears up a couple questions I had. Thanks!

Reply:
So [this is now possible in OAK](https://github.com/INCATools/ontology-access-kit/pull/656). I'm implementing now, but it is a bit slow; I estimate it'll take 104 minutes to complete. I'm going to make an GH issue about that. If you want, I can change this to SPARQL, I think that might be faster.

Reply:
## Results
@matentzn The script is basically finished, other than possibly addressing the performance issue highlighted above.

I had to change the robot template to be 1 edge per row rather than 1 term per row. Otherwise the 3rd column (subclass axiom sources) wouldn't make sense.

The table / file below is just a small snippet. What do you think?

|term_id      |parent_id    |source_ids                                         |FIELD4|FIELD5|FIELD6         |FIELD7 |
|-------------|-------------|---------------------------------------------------|------|------|---------------|-------|
|ID           |SC           |%                                                  |>     |A     |oboInOwl:source|SPLIT=&#124;|
|MONDO:0000004|MONDO:0002816|DOID:10493&#124;MONDO:Inferred                          |      |      |               |       |
|MONDO:0000005|MONDO:0021034|                                                   |      |      |               |       |
|MONDO:0000009|MONDO:0002243|DOID:2218&#124;MONDO:Redundant&#124;MONDO:cjm                |      |      |               |       |
|MONDO:0000009|MONDO:0002245|MONDO:Entailed&#124;MONDO:Redundant                     |      |      |               |       |
|MONDO:0000009|MONDO:0003847|                                                   |      |      |               |       |
|MONDO:0000014|MONDO:0001703|                                                   |      |      |               |       |
|MONDO:0000078|MONDO:0019796|Wikipedia:Acrocephalosyndactylia                   |      |      |               |       |
|MONDO:0000079|MONDO:0005298|                                                   |      |      |               |       |
|MONDO:0000082|MONDO:0005039|EFO:0004710                                        |      |      |               |       |
|MONDO:0000087|MONDO:0002320|NCIT:C116936                                       |      |      |               |       |
|MONDO:0000088|MONDO:0002259|MESH:D011629                                       |      |      |               |       |
|MONDO:0000110|MONDO:0018562|Orphanet:2695&#124;Orphanet:2695/inferred               |      |      |               |       |
|MONDO:0000118|MONDO:0019288|OMIMPS:179850&#124;https://orcid.org/0000-0002-4142-7153|      |      |               |       |

[sync-subclassof.robot.template.tsv.zip](https://github.com/monarch-initiative/mondo-ingest/files/12720880/sync-subclassof.robot.template.tsv.zip)

Reply:
This tables looks structurally correct, but the entire contents are wrong, as they are from what is already in Mondo.. The subclass sync should only include axioms from component ontologies!

So 

mondo-ingest.owl + sssom mappings in, and this table you are showing here out.. I am not sure where these specific rows are coming from but to my eyes these look like mondo.owl..

Reply:
Thanks, Nico. Part of the work has been unclear to me. My description of the file is:
> Create a robot template for purpose of syncing subclassof relations.

I know what each column is supposed to be. Maybe you can give me a sentence or few summary describing what the table is supposed to be and how it will be used?

There is no `sssom` input. The only input is an ontology file, and I'm using `merged.owl`. I think I never went over which input to use with you. Maybe if I swap this with `mondo-ingest.owl` it will work? I'll try that and post back here.

Alternatively, is what you want supposed to come from the slurp (TSVs?)? E.g. the newly proposed Mondo IDs that don't exist yet, their parents, and their sources?

Reply:
Nico responded with thorough explanation [in the issue](https://github.com/monarch-initiative/mondo-ingest/issues/92#issuecomment-1741773769).